File: src/main/java/com/nuutrai/reactor/entity/impl/cell/DoubleUraniumCellEntity.java
package com.nuutrai.reactor.entity.impl.cell;

import com.nuutrai.reactor.entity.Sellable;
import org.bukkit.Location;
import org.bukkit.Material;
import org.bukkit.entity.Player;

import static com.nuutrai.reactor.Reactor.logger;

public class DoubleUraniumCellEntity extends Sellable {

    public DoubleUraniumCellEntity(Player player, Location position) {
        super("uranium_double", Material.EMERALD_BLOCK, player, position);
    }

    public DoubleUraniumCellEntity() {
        super("uranium_double", Material.EMERALD_BLOCK);
    }

    @Override
    public void tick() {
        logger.info("Cell at " + this.getPosition().toLocation().getBlockX() + ", " + this.getPosition().toLocation().getBlockY() + ", " + this.getPosition().toLocation().getBlockZ() + " was ticked");
    }

    @Override
    public double getSellAmount() {
        return 0;
    }

    @Override
    public void sell() {

    }

    @Override
    public Sellable clone() {
        return new DoubleUraniumCellEntity();
    }

}

File: src/main/kotlin/com/nuutrai/reactor/entity/impl/cell/DoubleUraniumCellEntity.kt
package com.nuutrai.reactor.entity.impl.cell

import com.nuutrai.reactor.entity.Pairable
import com.nuutrai.reactor.entity.Sellable
import com.nuutrai.reactor.entity.lang.CellEntity
import com.nuutrai.reactor.util.MultiTypeMap
import org.bukkit.Material

class DoubleUraniumCellEntity : CellEntity("uranium_double", Material.EMERALD_BLOCK), Pairable {
	override fun tick(neighbours: Array<Sellable?>, params: MultiTypeMap?) {
		TODO()
	}

	override val sellAmount: Double
		get() = 0.0

	override fun sell() {
	}

	override fun clone(): Sellable {
		return DoubleUraniumCellEntity()
	}
}
